File: solutions/go/hamming/1/hamming.go
package hamming

import (
	"errors"
)

func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return 0, errors.New("strands are not the same length")
	}

	var distance int

	for i := 0; i < len(a); i++ {
		if a[i] != b[i] {
			distance++
		}
	}

	return distance, nil
}

File: solutions/csharp/hamming/1/Hamming.cs
public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand)
    {
        if (firstStrand.Length != secondStrand.Length)
        {
            throw new ArgumentException();
        }

        int distance = firstStrand.Length;

        for (var i = 0; i < firstStrand.Length; i++)
        {
            if (firstStrand[i] == secondStrand[i])
            {
                distance--;
            }
        }

        return distance;
    }
}